enterprise/app/code/code.tsx

```diff
@@ -466,6 +465,54 @@ export default class CodeComponent extends React.Component<Props, State> {
     return node?.type === "tree";
   }
 
+  async modelForBytestreamUrl(bytestreamURL: string, invocationID: string, path: string, zip?: string) {
+    let model = getOrCreateModel(path + "-error", "File not found in cache.");
+    await rpcService
+      .fetchBytestreamFile(bytestreamURL, invocationID, "text", { zip })
+      .then((result) => {
+        model = getOrCreateModel(path, result);
+      })
+      .catch(() => {});
+    return model;
+  }
+
+  showBytestreamCompare() {
+    const bytestreamURL = this.props.search.get("bytestream_url") || "";
+    const invocationID = this.props.search.get("invocation_id") || "";
+    const zip = this.props.search.get("z") || undefined;
+    let filename = this.props.search.get("filename") || "file";
+
+    const compareBytestreamURL = this.props.search.get("compare_bytestream_url") || "";
+    const compareInvocationID = this.props.search.get("compare_invocation_id") || "";
+    let compareFilename = this.props.search.get("compare_filename") || "";
+
+    let modelA = this.modelForBytestreamUrl(bytestreamURL, invocationID, filename, zip);
+    let modelB = this.modelForBytestreamUrl(compareBytestreamURL, compareInvocationID, "diff-" + compareFilename, zip);
+
+    Promise.all([modelA, modelB]).then((m) => {
+      if (!this.diffEditor) {
+        this.diffEditor = monaco.editor.createDiffEditor(this.diffViewer.current!);
+      }
+      let diffModel = { original: m[0], modified: m[1] };
```

Comment:
idk if this is clearer or not but you could avoid `Promise.all` and `m[0]` `m[1]`  by using await here:

```suggestion
      let diffModel = { original: await modelA, modified: await modelB };
```

Reply:
Good call, done!